Remove react-bootstrap and bootstrap Completely

SudharakaP opened the pull request:
This migrates all components that involves react-bootstrap and bootstrap and completely removes those dependencies.

Resolves https://github.com/opencollective/opencollective/issues/3223

# Screenshots

Here's a screenshot from before and after in the collective edit section; I checked most of the sections there and they seem to work okay. Since this is a larger change if you do see any small discrepancies feel free to let me know. 😄 

<details>
<summary>Desktop</summary>

![image](https://user-images.githubusercontent.com/12435965/102582979-97ceba80-40b8-11eb-9b35-4079b4878c72.png)
</details>

<details>
<summary>Mobile</summary>

![Screen Shot 2020-12-17 at 22 39 17](https://user-images.githubusercontent.com/12435965/102583041-b7fe7980-40b8-11eb-9735-fd41b9e7324a.png)
</details>

Betree reviewed (commented):
First of all congrats @SudharakaP. This is a huge change and it's almost impossible to catch all the side effects that removing such a global library can have. Overall, 90% is already covered and works great.

For the last 10%, I spotted a few design changes that we should iterate on before merging.

- Some inputs on the settings page seems smaller in height. Looking at the styles, it seems they have a `3rem` height defined, I'm not sure where this is coming from?

![image](https://user-images.githubusercontent.com/1556356/102864644-cb04a700-4434-11eb-8ade-bbfa2a0d0a51.png)

- It seems that the styles for `Hr` have changed:
![image](https://user-images.githubusercontent.com/1556356/102865367-ef14b800-4435-11eb-93e3-793145d6788b.png)
![image](https://user-images.githubusercontent.com/1556356/102865384-f4720280-4435-11eb-9707-0a26f3cd6b5c.png)

- On organization > Edit payment methods, the monthly limit per member input is too big and it seems the label isn't going to a new line:
![image](https://user-images.githubusercontent.com/1556356/102865543-30a56300-4436-11eb-8a9a-8a55b70a90fb.png)

- Another example of a label not going to a new line as it should:
![image](https://user-images.githubusercontent.com/1556356/102865678-6ba79680-4436-11eb-8547-6d269ff6f077.png)

- On `/:slug/edit/export`, the global styles for code blocks are gone. Not a blocker, but keping the defaults could be nice.
![image](https://user-images.githubusercontent.com/1556356/102866029-e96ba200-4436-11eb-8bb6-d654d2523097.png)
![image](https://user-images.githubusercontent.com/1556356/102866007-e1abfd80-4436-11eb-913c-031d31085d0e.png)

SudharakaP commented:
@Betree : I've addressed all the things you mentioned I think and it seems that I've missed some minor bootstrap stuff which I corrected. I've not interactively re-based here so that you can review the new changes more easily; they are contained in 8ad5326. So eventually the PR will need squashing. 😄

Betree commented on components/AddFundsSourcePicker.js:
```diff
@@ -56,28 +56,18 @@ class AddFundsSourcePickerForUser extends React.Component {
-            </option>
-          ))}
-        </FormControl>
+        <StyledSelect id="sourcePicker" placeholder="select" options={hostOptions} onChange={this.props.onChange} />
```
Maybe we could title case the placeholder. Also, this empty entry doesn't make much sense since the field is required, I think we should remove it. Finally, we should put `isSearchable={false}` on the select since there are only a few entries.

![image](https://user-images.githubusercontent.com/1556356/104162552-1dfaca00-53f5-11eb-9c19-019207152ee5.png)

SudharakaP replied:
@Betree : Thanks. I've corrected this. Please take a look and let me know if you see any other issues. 👍🏽

Betree commented on components/AddFundsSourcePicker.js:
```diff
@@ -56,28 +56,23 @@ class AddFundsSourcePickerForUser extends React.Component {
+          placeholder="Select"
+          options={hostOptions}
+          isSearchable={false}
+          onChange={this.props.onChange}
```
This will pass the full option (`{ label, value }`) to `onChange`. As a result, I'm getting this error when I try to submit:
![image](https://user-images.githubusercontent.com/1556356/104707466-b8704b80-571c-11eb-936e-418a39b6b10d.png)

SudharakaP replied:
@Betree : Thanks for pointing it out. I think I've made a mistake here. I've corrected it; please take a look. 😄

Betree reviewed (approved):
Working great! We need to resolve the conflict on `package.json`, but for the rest I think we're ready to merge this.

SudharakaP commented:
> Working great! We need to resolve the conflict on `package.json`, but for the rest I think we're ready to merge this.

@Betree : You are welcome. Conflict is resolved. 👍🏽
